packages/flutter/test/material/expansion_tile_test.dart

```diff
@@ -256,4 +256,99 @@ void main() {
   });
+
+  testWidgets('ExpansionTile alignment test', (WidgetTester tester) async {
+    const Key childKey = Key('key');
+    await tester.pumpWidget(MaterialApp(
+      home: Material(
+        child: Center(
+          child: ExpansionTile(
+            title: const Text('title'),
+            alignment: Alignment.centerLeft,
+            children: <Widget>[
+              Container(height: 100, width: 100),
+              Container(height: 100, width: 80, key: childKey,)
+            ],
+          ),
+        ),
+      ),
+    ));
+
+    await tester.tap(find.text('title'));
+    await tester.pumpAndSettle();
+
+    final Rect columnRect = tester.getRect(find.byType(Column).last);
+    final Rect secondChild = tester.getRect(find.byKey(childKey));
+
+    expect(columnRect.left, 0.0);
+    // The width of the Column is the maximum width of the children. The maximum
+    // width being 100.0, the offset of the right edge of Column from X-axis should
+    // be 100.0.
+    expect(columnRect.right, 100.0);
+
+    // The alignment doesn't define the position of the children inside the Column.
+    // Considering the default value for CrossAxisAlignment is CrossAxisAlignment.center,
+    // the offset of the left edge of second Container from X-axis should be greater
+    // than 0.
+    expect(secondChild.left, greaterThan(0.0));
+    expect(secondChild.right, lessThan(100.0));
+  });
+
+  testWidgets('ExpansionTile crossAxisAlignment test', (WidgetTester tester) async {
+    const Key child0Key = Key('child0');
+    const Key child1Key = Key('child1');
+    await tester.pumpWidget(MaterialApp(
+      home: Material(
+        child: Center(
+          child: ExpansionTile(
+            title: const Text('title'),
+            alignment: Alignment.centerRight,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              Container(height: 100, width: 100, key: child0Key,),
+              Container(height: 100, width: 80, key: child1Key,)
+            ],
+          ),
+        ),
+      ),
+    ));
+
+    await tester.tap(find.text('title'));
+    await tester.pumpAndSettle();
+
+    final Rect columnRect = tester.getRect(find.byType(Column).last);
+    final Rect child0Rect = tester.getRect(find.byKey(child0Key));
+    final Rect child1Rect = tester.getRect(find.byKey(child1Key));
+
+    expect(columnRect.right, 800.0);
+    // The width of the Column is the maximum width of the children. The maximum
+    // width being 100.0, the offset of the left edge of Column from X-axis should
+    // be 700.0.
+    expect(columnRect.left, 700.0);
+
+    // Considering the value of CrossAxisAlignment is CrossAxisAlignment.start,
+    // the offset of the left edge of both the children from X-axis should be 700.0.
+    expect(child0Rect.left, 700.0);
+    expect(child1Rect.left, 700.0);
+  });
+
+  testWidgets('ExpansionTile crossAxisAlignment.baseline', (WidgetTester tester) async {
```

Comment:
Could we also add a test testing that setting `textBaseline` causes a change in UI as well?

Reply:
@shihaohong  I can't think of a case where `textBaseline` is doing any changes in the UI. Usually, I use `textBaseline` for Row to align the baseline of different text. 

Can you provide any example for the `Column`?

Reply:
Hmm, that's true. It then feels odd to even introduce `textBaseline` here in this case since it would be confusing to provide this as an option. Should we then remove `textBaseline` and instead introduce an assert that prevents the use of `CrossAxisAlignment.textBaseline`?

Reply:
> 
> 
> Hmm, that's true. Sorry I hadn't thought about this a little more clearly. It then feels odd to even introduce `textBaseline` here in this case since it would be confusing to provide this as an option. Should we then remove `textBaseline` and instead introduce an assert that prevents the use of `CrossAxisAlignment.textBaseline`? If users come up with a use case for or a way to use `CrossAxisAlignment.textBaseline`, we can simply remove the assert and introduce `textBaseline` at that point.

That's a good suggestion. I have removed the `expandedTextbaseline` and added an assertion.